src/zkevm_specs/evm/execution/sdiv_smod.py

```diff
@@ -0,0 +1,104 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ...util import FQ, RLC
+
+
+def sdiv_smod(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    pop1 = instruction.stack_pop()
+    pop2 = instruction.stack_pop()
+    push = instruction.stack_push()
+
+    (a, b, c, d) = gen_witness(instruction, opcode, pop1, pop2, push)
+    check_witness(instruction, a, b, c, d)
+
+    instruction.step_state_transition_in_same_context(
+        opcode,
+        rw_counter=Transition.delta(3),
+        program_counter=Transition.delta(1),
+        stack_pointer=Transition.delta(1),
+    )
+
+
+def check_witness(instruction: Instruction, a: RLC, b: RLC, c: RLC, d: RLC):
+    a_is_zero = instruction.word_is_zero(a)
+    b_is_zero = instruction.word_is_zero(b)
+    c_is_zero = instruction.word_is_zero(c)
+
+    a_is_neg = instruction.word_is_neg(a)
+    b_is_neg = instruction.word_is_neg(b)
+    c_is_neg = instruction.word_is_neg(c)
+    d_is_neg = instruction.word_is_neg(d)
+
+    a_abs = instruction.abs_word(a)
+    b_abs = instruction.abs_word(b)
+    c_abs = instruction.abs_word(c)
+    d_abs = instruction.abs_word(d)
+
+    # Constrain abs(remainder) < abs(divisor) when divisor != 0.
+    lt, _ = instruction.compare_word(c_abs, b_abs)
+    instruction.constrain_zero((1 - lt) * (1 - b_is_zero))
+
+    # Constrain overflow == 0.
+    overflow = instruction.mul_add_words(a_abs, b_abs, c_abs, d_abs)
+    instruction.constrain_zero(overflow)
+
+    # Constrain sign(dividend) == sign(remainder) when quotient, divisor and
+    # remainder are all non-zero.
+    condition = (1 - a_is_zero) * (1 - b_is_zero) * (1 - c_is_zero)
+    instruction.constrain_equal(d_is_neg * condition, c_is_neg * condition)
+
+    # The dividend is signed overflow when `-(1 << 255) // -1 = (1 << 255)`.
+    d_is_signed_overflow = instruction.word_is_neg(d_abs)
```

Comment:
it seems strange that an absolute value could be negative?

Reply:
Same as above test case.